hypershift-operator/controllers/platform/aws/controller_test.go

```diff
@@ -54,43 +54,77 @@ func (f *fakeEC2Client) ModifyVpcEndpointServicePermissions(in *ec2.ModifyVpcEnd
 }
 
 func TestReconcileAWSEndpointServiceStatus(t *testing.T) {
-	elbClient := &fakeElbv2Client{out: &elbv2.DescribeLoadBalancersOutput{LoadBalancers: []*elbv2.LoadBalancer{{
-		LoadBalancerArn: aws.String("lb-arn"),
-		State:           &elbv2.LoadBalancerState{Code: aws.String(elbv2.LoadBalancerStateEnumActive)},
-	}}}}
-
-	infra := &configv1.Infrastructure{
-		ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
-		Status:     configv1.InfrastructureStatus{InfrastructureName: "management-cluster-infra-id"},
-	}
-	client := fake.NewClientBuilder().WithScheme(hyperapi.Scheme).WithObjects(infra).Build()
-
-	ec2Client := &fakeEC2Client{
-		createOut: &ec2.CreateVpcEndpointServiceConfigurationOutput{ServiceConfiguration: &ec2.ServiceConfiguration{ServiceName: aws.String("ep-service")}},
-		permsOut:  &ec2.DescribeVpcEndpointServicePermissionsOutput{},
-	}
-
-	roleARN := "fakeRoleARN"
-	r := AWSEndpointServiceReconciler{
-		Client: client,
-		controlPlaneOperatorRoleARNFn: func(ctx context.Context, hc *hyperv1.HostedCluster) (string, error) {
-			return roleARN, nil
+	tests := []struct {
+		name                        string
+		additionalAllowedPrincipals []string
+	}{
+		{
+			name:                        "no additional principals",
+			additionalAllowedPrincipals: []string{},
+		},
+		{
+			name:                        "additional principals",
+			additionalAllowedPrincipals: []string{"additional1", "additional2"},
```

Comment:
it'd be nice to validate we can go from more additional to less.

Reply:
Oh, good point, I can add this logic/test

Reply:
Should be done! Tweaked the test case slightly to test for adding and removing principals